zcash_keys/src/address.rs

```diff
@@ -293,20 +339,24 @@ mod tests {
             Some(fvk.address_at(0u32, orchard::keys::Scope::External))
         };
 
+        #[cfg(feature = "sapling")]
         let sapling = {
             let extsk = sapling::spending_key(&[0; 32], 0, AccountId::ZERO);
             let dfvk = extsk.to_diversifiable_full_viewing_key();
             Some(dfvk.default_address().1)
         };
 
-        let transparent = { None };
+        let transparent = None;
 
-        #[cfg(feature = "orchard")]
+        #[cfg(all(feature = "orchard", feature = "sapling"))]
         let ua = UnifiedAddress::from_receivers(orchard, sapling, transparent).unwrap();
 
-        #[cfg(not(feature = "orchard"))]
+        #[cfg(all(not(feature = "orchard"), feature = "sapling"))]
```

Comment:
We need an additional case for `all(feature = "orchard", not(feature = "sapling"))`.